In [1]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)

In [9]:
import os

print("Current Folder:")
print(os.getcwd())

print("\nParent Folder Files:")
print(os.listdir(".."))

Current Folder:
c:\Users\lathi\OneDrive\Desktop\Indian-House-Price-Prediction\notebooks

Parent Folder Files:
['app.py', 'dataset', 'images', 'model', 'notebooks', 'README.md', 'reports', 'requirements.txt', 'train_model.py']


In [11]:
df = pd.read_csv("../dataset/bengaluru_house_prices.csv")
df.head()

,area_type,availability,location,size,society,total_sqft,bath,balcony,price
0,Super built-up Area,19-Dec,Electronic City Phase II,2 BHK,Coomee,1056,2.0,1.0,39.07
1,Plot Area,Ready To Move,Chikka Tirupathi,4 Bedroom,Theanmp,2600,5.0,3.0,120.00
2,Built-up Area,Ready To Move,Uttarahalli,3 BHK,NaN,1440,2.0,3.0,62.00
3,Super built-up Area,Ready To Move,Lingadheeranahalli,3 BHK,Soiewre,1521,3.0,1.0,95.00
4,Super built-up Area,Ready To Move,Kothanur,2 BHK,NaN,1200,2.0,1.0,51.00


In [12]:
print("Rows :", df.shape[0])
print("Columns :", df.shape[1])

Rows : 13320
Columns : 9


In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13320 entries, 0 to 13319
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   area_type     13320 non-null  object 
 1   availability  13320 non-null  object 
 2   location      13319 non-null  object 
 3   size          13304 non-null  object 
 4   society       7818 non-null   object 
 5   total_sqft    13320 non-null  object 
 6   bath          13247 non-null  float64
 7   balcony       12711 non-null  float64
 8   price         13320 non-null  float64
dtypes: float64(3), object(6)
memory usage: 936.7+ KB


In [14]:
df.dtypes

area_type        object
availability     object
location         object
size             object
society          object
total_sqft       object
bath            float64
balcony         float64
price           float64
dtype: object

In [15]:
df.isnull().sum()

area_type          0
availability       0
location           1
size              16
society         5502
total_sqft         0
bath              73
balcony          609
price              0
dtype: int64

In [16]:
missing = (df.isnull().sum() / len(df)) * 100

missing.sort_values(ascending=False)

society         41.306306
balcony          4.572072
bath             0.548048
size             0.120120
location         0.007508
area_type        0.000000
availability     0.000000
total_sqft       0.000000
price            0.000000
dtype: float64

In [17]:
print("Duplicate Rows :", df.duplicated().sum())

Duplicate Rows : 529


In [18]:
df.describe()

,bath,balcony,price
count,13247.000000,12711.000000,13320.000000
mean,2.692610,1.584376,112.565627
std,1.341458,0.817263,148.971674
min,1.000000,0.000000,8.000000
25%,2.000000,1.000000,50.000000
50%,2.000000,2.000000,72.000000
75%,3.000000,2.000000,120.000000
max,40.000000,3.000000,3600.000000


In [19]:
df.describe(include="object")

,area_type,availability,location,size,society,total_sqft
count,13320,13320,13319,13304,7818,13320
unique,4,81,1305,31,2688,2117
top,Super built-up Area,Ready To Move,Whitefield,2 BHK,GrrvaGr,1200
freq,8790,10581,540,5199,80,843


In [20]:
for col in df.columns:
    print(f"{col} : {df[col].nunique()}")

area_type : 4
availability : 81
location : 1305
size : 31
society : 2688
total_sqft : 2117
bath : 19
balcony : 4
price : 1994


# Observations

- The dataset contains **13,320 rows** and **9 columns**.

- The dataset has both **numerical** and **categorical** features.

### Numerical Features
- total_sqft
- bath
- balcony
- price

### Categorical Features
- area_type
- availability
- location
- size
- society

### Missing Values
- society contains a large number of missing values.
- balcony contains missing values.
- bath contains a few missing values.

### Initial Conclusion
- The dataset requires preprocessing before training machine learning models.
- Missing values and inconsistent data formats will be handled in the next notebook.
- The dataset is suitable for an Indian House Price Prediction project.